app/controllers/passwords_controller.rb

```diff
@@ -32,8 +32,21 @@ def mfa_edit
     end
   end
 
+  def verify
+    if verify_user
+      session[:verification] = Time.current + 10.minutes
+      redirect_to verify_password_params[:redirect_uri]
+    else
+      render_not_found
```

Comment:
Please redirect to for `"passwords/verify"` with a flash message that the password was incorrect. 
Also, please make sure we are checking signed in user for this action.